src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -328,6 +332,91 @@ public Response uningestDatafile(@PathParam("id") String id) {
         }
 
     }
+    
+    // reingest attempts to queue an *existing* DataFile 
+    // for tabular ingest. It can be used on non-tabular datafiles; to try to 
+    // ingest a file that has previously failed ingest, or to ingest a file of a
+    // type for which ingest was not previously supported. 
+    // It will also be possible to *force* reingest of a datafile that's already
+    // ingested as Tabular data; for example, to address a bug that has been 
+    // found in an ingest plugin. 
+    
+    @Path("{id}/reingest")
+    @POST
+    public Response reingest(@PathParam("id") String id) {
+
+        AuthenticatedUser u;
+        try {
+            u = findAuthenticatedUserOrDie();
+            if (!u.isSuperuser()) {
+                return error(Response.Status.FORBIDDEN, "This API call can be used by superusers only");
+            }
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
+        }
+        
+        DataFile dataFile;
+        try {
+            dataFile = findDataFileOrDie(id);
+        } catch (WrappedResponse ex) {
+            dataFile = null;
```

Comment:
That was in case findDataFileOrDie() returns null, without throwing an exception. 
Which I guess it should never be the case. As the name of the method suggests. :)